### ISSUE TITLE
Task Tile

### ISSUE THREAD

**Issue by Santhisanduri** [Santhisanduri/O2i#2]
please give me project structure and  wmls site contacts  wants add in o2i db

**Comment by Santhisanduri** [comment]
Found some old convos and stuff, posting some finding soon re: notifications
Context: There was work over a year and a half ago to get wmls retired sooner, so questions were brought up on what dependencies still existed with O2I and I was asked this and I mentioned, naturally, notifications.
Gathering my thoughts and links, so here we go (edited) 
:raised_hands::skin-tone-2:
1

5 replies


Umesh Mandadapu
  [Wednesday at 3:34 PM](https://disneystudios.slack.com/archives/C071TGE5FPD/p1725489262563379?thread_ts=1725487854.862729&cid=C071TGE5FPD)
[@richly](https://disneystudios.slack.com/team/WSH9U0LHZ)
: Thanks, Are you planning to send links or attachments to this message


Richly Chheuy
  Wednesday at 3:52 PM
Since we're talking about Transfer Manager, we'll be looking at the https://gitlab.disney.com/O2I/o2i-transfer-manager-web repo.
In particular:
https://gitlab.disney.com/O2I/o2i-transfer-manager-web/-/blob/develop/src/main/java/com/disney/spt/receiving/process/step/AbstractSimpleNotificationProcessStep.java
https://gitlab.disney.com/O2I/o2i-transfer-manager-web/-/blob/develop/src/main/java/com/disney/spt/receiving/process/step/TransferToTargetNotificationProcessStep.java
TransferToTargetNotificationProcessStep extends AbstractSimpleNotificationProcessStep class above. Looking at buildNotificationTask()
Notice instantiations of wmls-related classes, e.g. wmlsService . This triggers the lookup to get site info and create the notification email. Probably can refactor the logic here to start and instead do a lookup via local db/data that matches siteId being passed through the workflow (if you're cloning that data from wmls db)
protected Task buildNotificationTask() {
		StorageSite site = wmlsService.getSite(getProcessData().getTargetLocationId());
		EmailTemplate template = notificationService.buildEmailTemplate(getConfigurationProperty(EMAIL_TEMPLATE, true));
		
		Set<NotificationContact> contacts = new HashSet<>();
		contacts.add(new NotificationContact(vendorAndContactService.getSptUserForUsername(getProcessData().getUserName())));
		contacts.add(new NotificationContact(site.getPrimaryContact()));
		contacts.addAll(wmlsService.getSiteNotificationContacts(site)); 
		contacts.addAll(notificationService.emailsToContacts(getProcessData().getAdditionalContacts()));
		
		Map<String, Object> emailParams = new HashMap<>();
		emailParams.put("targetDesc", wmlsService.getSite(getProcessData().getTargetLocationId()).getDescription());

		SendNotificationForReceivingRequestTask task = buildTask(SendNotificationForReceivingRequestTask.class);
		task.init(getProcessData(), emailParams, contacts, null, null, template);
		
		return task;
	}
Other related:
https://gitlab.disney.com/O2I/o2i-wmls-client/-/blob/develop/src/main/java/com/disney/spt/service/wmls/WmlsRestService.java
https://gitlab.disney.com/O2I/o2i-wmls-client/-/blob/develop/src/main/java/com/disney/spt/service/wmls/WmlsRestClient.java
Snapshot of convo thread from a long while back when asked about this and the stack trace of the error when WMLS db is inaccessible when generating the notification.
Can see which lines in source are being executed up until error. (edited) 
2 files
 
Screenshot 2024-09-04 at 3.50.40 PM.png


Screenshot 2023-03-16 at 3.10.28 PM (1).png


[3:55](https://disneystudios.slack.com/archives/C071TGE5FPD/p1725490544487659?thread_ts=1725487854.862729&cid=C071TGE5FPD)
Easier to deal with transfer manager web pipeline since that has been most recently changed, tested.


Richly Chheuy
  Wednesday at 4:05 PM
tl;dr I guess ultimate create your own custom class(es) to instead call local o2i db data.
Some examples where some o2i notifs build an email from user_account table I believe, such as for Transfer Manager problem emails. That is taking users who are part of spt_group_membership admin and apply to them to field:
https://gitlab.disney.com/O2I/o2i-transfer-manager-web/-/blob/develop/src/main/resources/transfermanager.properties
[https://gitlab.disney.com/O2I/o2i-transfer-manager-web/-/blob/develop/src/main/java/c[…]/spt/receiving/service/notification/NotificationService.java](https://gitlab.disney.com/O2I/o2i-transfer-manager-web/-/blob/develop/src/main/java/com/disney/spt/receiving/service/notification/NotificationService.java)
notification.problem.targetGroup
#Notification configuration
#used as a flag to disable all email when testing
notification.email.disableEmail = false
notification.email.server = sptmailrelay.studio.disney.com
notification.email.fromAddress = twds.transmgr@studio.disney.com
notification.email.fromName =  Disney Transfer Manager
notification.email.groupsToCopy = SPT-SoftwareNotifications
notification.email.addressesToCopy =
notification.email.subjectPrefix = (Dev)

notification.problem.targetGroup = ${admin.group}
notification.requestDetail.uri = http://foo.bar.baz/
(edited)


Richly Chheuy
  Wednesday at 4:34 PM
On the zoom call, https://gitlab.disney.com/O2I/o2i-email-service/-/blob/develop/emailapi-web/src/main/java/com/disney/spt/mail/controller/ApiController.java was shown, but no wmls classes are referenced, so I think the o2i-email-service is more of a way to better manage or handle email templating (when emails are being sent out) for various types of notifications (as we have to remember O2I being a beast of an app back in its heyday and different types of emails were sent out depending on the app or workflow). (edited)" please read this chat we have a new project for you guys. As you may have heard, WMLS is moving to Studio to be deprecated. O2i currently has dependencies within WMLS- particularly the table pertaining to the notification step of TM1 transfers wmls_site_contacts & also other tables holding the foreign keys within this. We will need to pull this table into the O2i db & also update the code configurations within the O2i repos. The first step is to clone down the [O2i repos](https://gitlab.disney.com/O2I?page=2) & try to locate where this table is in the code

**Comment by Santhisanduri** [comment]
Found some old convos and stuff, posting some finding soon re: notifications
Context: There was work over a year and a half ago to get wmls retired sooner, so questions were brought up on what dependencies still existed with O2I and I was asked this and I mentioned, naturally, notifications.
Gathering my thoughts and links, so here we go (edited)
🙌:skin-tone-2:
1

5 replies

Umesh Mandadapu
[Wednesday at 3:34 PM](https://disneystudios.slack.com/archives/C071TGE5FPD/p1725489262563379?thread_ts=1725487854.862729&cid=C071TGE5FPD)
[@richly](https://disneystudios.slack.com/team/WSH9U0LHZ)
: Thanks, Are you planning to send links or attachments to this message

Richly Chheuy
Wednesday at 3:52 PM
Since we're talking about Transfer Manager, we'll be looking at the https://gitlab.disney.com/O2I/o2i-transfer-manager-web repo.
In particular:
https://gitlab.disney.com/O2I/o2i-transfer-manager-web/-/blob/develop/src/main/java/com/disney/spt/receiving/process/step/AbstractSimpleNotificationProcessStep.java
https://gitlab.disney.com/O2I/o2i-transfer-manager-web/-/blob/develop/src/main/java/com/disney/spt/receiving/process/step/TransferToTargetNotificationProcessStep.java
TransferToTargetNotificationProcessStep extends AbstractSimpleNotificationProcessStep class above. Looking at buildNotificationTask()
Notice instantiations of wmls-related classes, e.g. wmlsService . This triggers the lookup to get site info and create the notification email. Probably can refactor the logic here to start and instead do a lookup via local db/data that matches siteId being passed through the workflow (if you're cloning that data from wmls db)
protected Task buildNotificationTask() {
StorageSite site = wmlsService.getSite(getProcessData().getTargetLocationId());
EmailTemplate template = notificationService.buildEmailTemplate(getConfigurationProperty(EMAIL_TEMPLATE, true));

	Set<NotificationContact> contacts = new HashSet<>();
	contacts.add(new NotificationContact(vendorAndContactService.getSptUserForUsername(getProcessData().getUserName())));
	contacts.add(new NotificationContact(site.getPrimaryContact()));
	contacts.addAll(wmlsService.getSiteNotificationContacts(site)); 
	contacts.addAll(notificationService.emailsToContacts(getProcessData().getAdditionalContacts()));
	
	Map<String, Object> emailParams = new HashMap<>();
	emailParams.put("targetDesc", wmlsService.getSite(getProcessData().getTargetLocationId()).getDescription());

	SendNotificationForReceivingRequestTask task = buildTask(SendNotificationForReceivingRequestTask.class);
	task.init(getProcessData(), emailParams, contacts, null, null, template);
	
	return task;
}
Other related:
https://gitlab.disney.com/O2I/o2i-wmls-client/-/blob/develop/src/main/java/com/disney/spt/service/wmls/WmlsRestService.java
https://gitlab.disney.com/O2I/o2i-wmls-client/-/blob/develop/src/main/java/com/disney/spt/service/wmls/WmlsRestClient.java
Snapshot of convo thread from a long while back when asked about this and the stack trace of the error when WMLS db is inaccessible when generating the notification.
Can see which lines in source are being executed up until error. (edited)
2 files

Screenshot 2024-09-04 at 3.50.40 PM.png

Screenshot 2023-03-16 at 3.10.28 PM (1).png

[3:55](https://disneystudios.slack.com/archives/C071TGE5FPD/p1725490544487659?thread_ts=1725487854.862729&cid=C071TGE5FPD)
Easier to deal with transfer manager web pipeline since that has been most recently changed, tested.

Richly Chheuy
Wednesday at 4:05 PM
tl;dr I guess ultimate create your own custom class(es) to instead call local o2i db data.
Some examples where some o2i notifs build an email from user_account table I believe, such as for Transfer Manager problem emails. That is taking users who are part of spt_group_membership admin and apply to them to field:
https://gitlab.disney.com/O2I/o2i-transfer-manager-web/-/blob/develop/src/main/resources/transfermanager.properties
[https://gitlab.disney.com/O2I/o2i-transfer-manager-web/-/blob/develop/src/main/java/c[…]/spt/receiving/service/notification/NotificationService.java](https://gitlab.disney.com/O2I/o2i-transfer-manager-web/-/blob/develop/src/main/java/com/disney/spt/receiving/service/notification/NotificationService.java)
notification.problem.targetGroup
#Notification configuration
#used as a flag to disable all email when testing
notification.email.disableEmail = false
notification.email.server = sptmailrelay.studio.disney.com
notification.email.fromAddress = [twds.transmgr@studio.disney.com](mailto:twds.transmgr@studio.disney.com)
notification.email.fromName = Disney Transfer Manager
notification.email.groupsToCopy = SPT-SoftwareNotifications
notification.email.addressesToCopy =
notification.email.subjectPrefix = (Dev)

notification.problem.targetGroup = ${admin.group}
notification.requestDetail.uri = http://foo.bar.baz/
(edited)

Richly Chheuy
Wednesday at 4:34 PM
On the zoom call, https://gitlab.disney.com/O2I/o2i-email-service/-/blob/develop/emailapi-web/src/main/java/com/disney/spt/mail/controller/ApiController.java was shown, but no wmls classes are referenced, so I think the o2i-email-service is more of a way to better manage or handle email templating (when emails are being sent out) for various types of notifications (as we have to remember O2I being a beast of an app back in its heyday and different types of emails were sent out depending on the app or workflow). (edited)" please read this chat we have a new project for you guys. As you may have heard, WMLS is moving to Studio to be deprecated. O2i currently has dependencies within WMLS- particularly the table pertaining to the notification step of TM1 transfers wmls_site_contacts & also other tables holding the foreign keys within this. We will need to pull this table into the O2i db & also update the code configurations within the O2i repos. The first step is to clone down the [O2i repos](https://gitlab.disney.com/O2I?page=2) & try to locate where this table is in the code""
